gnucash/gnome/gnc-plugin-page-report.cpp

```diff
@@ -1793,17 +1792,37 @@ gnc_plugin_page_report_export_cb (GSimpleAction *simple,
                                      "document object with export-string or "
                                      "export-error."));
         }
-        result = TRUE;
     }
     else
-        result = gnc_html_export_to_file (priv->html, filepath);
-
-    if (!result)
     {
-        const char *fmt = _("Could not open the file %s. "
-                            "The error is: %s");
-        gnc_error_dialog (parent, fmt, filepath ? filepath : "(null)",
-                          strerror (errno) ? strerror (errno) : "" );
+        SCM chart_report = scm_call_1 (scm_c_eval_string ("gnc:report-chart?"),
+                                       priv->cur_report);
+        SCM html_export = (scm_is_true (chart_report)) ? SCM_BOOL_T : SCM_BOOL_F;
+        SCM report_render_html = scm_c_eval_string ("gnc:report-render-html");
+        SCM apply_with_error_handling = scm_c_eval_string ("gnc:apply-with-error-handling");
+        SCM arg_list = scm_list_3 (priv->cur_report, SCM_BOOL_T, html_export);
+        SCM rendered = scm_call_2 (apply_with_error_handling, report_render_html, arg_list);
+        SCM document = SCM_CAR(rendered);
+        SCM err = SCM_CDR (rendered);
+
+        if (scm_is_false (document))
+        {
+            gnc_error_dialog (parent, "%s %s",
+                              _("Error rendering HTML:"),
+                              scm_to_utf8_string (err));
```

Comment:
This scm_to_utf8_string must be g_freed after use.

Reply:
Oopsy. Will fix.

Reply:
Fixed